lib/HLSL/DxilValidation.cpp

```diff
@@ -1658,7 +1658,9 @@ static void ValidateResourceDxilOp(CallInst *CI, DXIL::OpCode opcode,
   case DXIL::OpCode::CalculateLOD: {
     DxilInst_CalculateLOD lod(CI);
     Value *samplerHandle = lod.get_sampler();
-    if (GetSamplerKind(samplerHandle, ValCtx) != DXIL::SamplerKind::Default) {
+    if ((DXIL::CompareVersions(ValCtx.m_DxilMajor, ValCtx.m_DxilMinor, 1, 8) <
+         0) &&
```

Comment:
Not necessarily part of this change, but seeing how ugly this gets with clang-format makes me think that `ValidationContext` should perhaps have an `isDxilVersionSM6XPlus()` function. 

I do think we should still check that the sampler kind isn't invalid for dxil version 1.8+

Reply:
Bring back check for 1.8+.